gcm/gcm.py

```diff
@@ -336,12 +336,21 @@ def json_request(self, registration_ids, data=None, collapse_key=None,
         )
 
         backoff = self.BACKOFF_INITIAL_DELAY
+        unavailable_err = None
         for attempt in range(retries):
+            payload = self.construct_payload(
```

Comment:
I see two problems with this line:
1) The payload is already constructed outside the loop the first time around and you are constructing it again
2) The construct_payload method call inside the for loop doesn't take the same parameters as the one outside the loop